pkg/hash_cache.go

```diff
@@ -38,7 +41,37 @@ func NewTargetHashCache(context map[gazelle_label.Label]map[Configuration]*analy
 }
 
 func isConfiguredRuleInputsSupported(releaseString string) bool {
-	// Disabled until https://github.com/bazelbuild/bazel/issues/17916 is resolved
+	releasePrefix := "release "
+	explanation := " - assuming cquery does not support configured rule inputs (which is supported from bazel 7), which may lead to over-estimates of affected targets"
+	if !strings.HasPrefix(releaseString, releasePrefix) {
+		log.Printf("Bazel wasn't a released version%s", explanation)
+		return false
+	}
+	v, err := version.NewVersion(releaseString[len(releasePrefix):])
+	if err != nil {
+		log.Printf("Failed to parse Bazel version %q: %s", releaseString, explanation)
+		return false
+	}
+	if v.GreaterThanOrEqual(version.Must(version.NewVersion("7.0.0"))) {
+		return true
+	}
+
+	// Support 7.0.0 pre-releases which include https://github.com/bazelbuild/bazel/commit/9575c7fa86458717c154f08ff8a7f7aff000f6ae
+	segments := v.Segments()
+	if len(segments) == 3 && segments[0] == 7 && segments[1] == 0 && segments[2] == 0 {
+		pre := v.Prerelease()
+		prereleasePrefix := "pre."
+		if strings.HasPrefix(pre, prereleasePrefix) {
+			dateString := strings.Split(pre[len(prereleasePrefix):], ".")[0]
+			dateNumber, err := strconv.Atoi(dateString)
+			if err == nil {
+				if dateNumber >= 20230628 {
+					return true
+				}
+			}
+		}
+	}
+
 	return false
```

Comment:
Given your needs, the following simplification might work:

```suggestion
	return v.GreaterThanOrEqual(version.Must(version.NewVersion("7.0.0-pre.20230628.2")))
```

Given the released versions of Bazel [here](https://github.com/bazelbuild/bazel/releases), here is the test I used for this:

```go
func Test_isConfiguredRuleInputsSupported(t *testing.T) {
	testCases := map[string]bool{
		"release 6.3.1":                false,
		"release 7.0.0-pre.20230530.3": false,
		"release 7.0.0-pre.20230816.3": true,
		"release 7.0.0":                true,
	}

	for version, want := range testCases {
		t.Run(version, func(t *testing.T) {
			got := isConfiguredRuleInputsSupported(version)
			if want != got {
				t.Fatalf("Wrong absolute path: want %v got %v", want, got)
			}
		})
	}
}
```

Reply:
Nice! Thanks!